### ISSUE TITLE
Acting buggy on another use case

### ISSUE THREAD

**Issue by felixsanz** [arve0/markdown-it-implicit-figures#12]
Hey! :grimacing: I've found another use case where the plugin fails to parse correctly.

The markdown content is this:

```
## Header

[link](http://www.example.com)

## Header

[link](http://www.example.com)
```

Returns this:

```
<h2>Header</h2>
<figure><a href="http://www.example.com">link</a></figure>
<h2>Header</h2>
<figure><a href="http://www.example.com">link</a></figure>
```

Example: https://jsfiddle.net/jzyag4pv/4/

**Comment by arve0** [comment]
Dammit I'm sloppy reviewing PRs. Also related to #10. Fix in 2 minutes.

**Comment by arve0** [comment]
Fixed in v0.3.2
